lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,18 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      basic = super
+      # The inherited equality is always accepted, so use it if valid.
+      return basic if basic || ![:mtime, :ctime].include?(checksum_type)
+      return false unless current && current != :absent && desired && desired != :absent
+      begin
+        DateTime.parse(current) >= DateTime.parse(desired)
```

Comment:
This accepts a md5 SHA (why?) as a valid date, so I'm modifying it a bit. I'm also trying to add tests that have positive validation that an update will occur when ctime/mtime are different, but finding it difficult to stub ctime/mtime the way I need to.